### ISSUE TITLE
Scoped packages not supported; unknown package type

### ISSUE THREAD

**Issue by jrop** [watilde/dep#9]
### Steps to reproduce behavior

The following dependencies creates a few errors:

**package.json**
```json
{
	"dependencies": {
		"minimist": "^1.2.0",
		"perplex": "^0.10.2"
	},
	"devDependencies": {
		"@types/node": "^8.0.17",
		"@types/tape": "^4.2.30",
		"ts-node": "^3.3.0"
	}
}
```

First, the scoped packages `@types/node` and `@types/tape` are not mapped to valid URLs in `lib/install/resolver/fetchers/registry.js`: the `/` character needs to be replaced with `%2f` in order for the registry to take the package name correctly:

```js
    const options = {
      url: npmrc.registry + name.replace(/\//g, '%2f') + '/' + spec,
      //                         ^^^^^^^^^^^^^^^^^^^^^
      headers: {
        'User-Agent': npmrc.userAgent
      }
    }
```
Also curious, is even when I add that code, this particular dependency tree still does not install, failing with the error:

```sh
$ dep i
Resolving dependencies
Installing dependencies
TypeError: Cannot read property 'then' of undefined
    at installer (/usr/local/lib/node_modules/dep/lib/install/installer.js:43:8)
    at Promise (/usr/local/lib/node_modules/dep/lib/install/installer.js:77:45)
    at Promise (<anonymous>)
    at Object.keys.map (/usr/local/lib/node_modules/dep/lib/install/installer.js:77:12)
    at Array.map (native)
    at module.exports (/usr/local/lib/node_modules/dep/lib/install/installer.js:76:28)
    at Promise.all.then (/usr/local/lib/node_modules/dep/lib/install.js:48:19)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:169:7)%
```

### Expected behavior

Installation without errors

### Actual behavior

The errors above.

**Comment by watilde** [comment]
Hello! Thanks for reporting the issue. This a bug caused by what you said. What we need to do is using the `escapedName`provided by [npm-package-arg](https://www.npmjs.com/package/npm-package-arg).

I will work on this from now on.

**Comment by watilde** [comment]
The patch was released as `v0.14.2` and this issue should be fixed. Thanks again for reporting the issue with the full information<3

Cheer!
